featuretools/feature_base/feature_base.py

```diff
@@ -643,19 +646,18 @@ class TransformFeature(FeatureBase):
     def __init__(self, base_features, primitive, name=None):
         # Any edits made to this method should also be made to the
         # new_class_init method in make_trans_primitive
-        if hasattr(base_features, '__iter__'):
-            base_features = [_check_feature(bf) for bf in base_features]
+        base_features = _check_feature(base_features)
+        if isinstance(base_features, list):
             msg = "all base features must share the same dataframe"
             assert len(set([bf.dataframe_name for bf in base_features])) == 1, msg
         else:
-            base_features = [_check_feature(base_features)]
+            base_features = [base_features]
```

Comment:
Since this code block gets used in multiple classes and we already moved some logic into `_check_feature`, how about we rename it to `_validate_base_features` and move all this logic in there.

So on the classes it would look like

```python
base_features = _validate_base_features(base_features)
```
and life goes on

Reply:
Sounds good, will update

Reply:
✅  39989675b01117d11a313abc10af8fbb1c6c2fd0